src/Middleware/Authentication/AppInstallationAuthProvider.cs

```diff
@@ -0,0 +1,18 @@
+using Microsoft.Kiota.Abstractions.Authentication;
+
+namespace GitHub.Octokit.Client.Authentication;
+
+
+/// <summary>
+/// Represents an authentication provider for app installations.
+/// This class is a concrete implementation of <see cref="BaseBearerTokenAuthenticationProvider"/>.
+/// This is beneficial for dev ergonomics - where BaseBearerTokenAuthenticationProvider
```

Comment:
You should be able to see how this is relevant in the CLI.  i.e.

If this wasn't in place, you'd have this:

```
var atp = new AppInstallationTokenProvider(clientId, rsa, installationId, githubAppTokenProvider);
var aa = RequestAdapter.Create(new BaseBearerTokenAuthenticationProvider(aiAccessTokenProvider));
```

Instead of this:

```
var atp = new AppInstallationTokenProvider(clientId, rsa, installationId, githubAppTokenProvider);
var aa = RequestAdapter.Create(new AppInstallationAuthProvider(aiAccessTokenProvider));
```

Ergonomically, using the base class just feels clumsy and if we ever need to add anything to the way we do App Installation Auth then we can easily do it here.  The latter is less of a concern but it still protects the SDK from breaking changes in Auth if the need ever arises.

Reply:
Got it, thank you for the explanation!